sigstore/sign.py

```diff
@@ -308,31 +331,46 @@ def signer(
         yield Signer(identity_token, self, cache)
 
 
-class SigningResult(BaseModel):
+@dataclass
+class SigningResult:
```

Comment:
Flagging for review: the changes here indicate that we should probably just drop this type entirely and switch to `Bundle` as our "signing result" type. It'd be strictly equivalent and would save us a layer of types.

Reply:
I think consolidating to using `Bundle` is a good idea. I'm happy to follow up with a change to do that to keep things simple in this PR.

Reply:
This sounds right

Reply:
Thanks both! @tetsuo-cpp, I'll assign you a follow-up for removing `SigningResult`.